GoogleDataTransport/GDTCORTests/Unit/GDTCORFlatFileStorageTest.m

```diff
@@ -863,6 +830,167 @@ - (void)testStorageSizeWithCallback {
   [self waitForExpectations:@[ expectation ] timeout:10.0];
 }
 
+/** Tests generating the next batchID. */
+- (void)testNextBatchID {
+  BOOL originalContinueAfterFailure = self.continueAfterFailure;
+  self.continueAfterFailure = NO;
+  NSNumber *expectedBatchID = @0;
+  NSNumber *batchID = [[GDTCORFlatFileStorage sharedInstance] nextBatchID];
+  XCTAssertNotNil(batchID);
+  XCTAssertEqualObjects(batchID, expectedBatchID);
+
+  expectedBatchID = @1;
+  batchID = [[GDTCORFlatFileStorage sharedInstance] nextBatchID];
+  XCTAssertNotNil(batchID);
+  XCTAssertEqualObjects(batchID, expectedBatchID);
+
+  for (int i = 0; i < 1000; i++) {
+    NSNumber *batchID = [[GDTCORFlatFileStorage sharedInstance] nextBatchID];
```

Comment:
It may be useful to have a test for concurrent calling of `nextBatchID` to assert its thread safety.

Reply:
I think calling it 1000 times almost as fast as possible is sufficient (It regularly failed when it wasn't thread-safe during development).

Reply:
The current test does the calls one by one and will not fail if we remove semaphore from `nextBatchID` implementation. I was thinking of using e.g. `dispatch_async` on a concurrent queue e.g. like [here](https://github.com/firebase/firebase-ios-sdk/blob/5f370d6395c51a8616157d7b2cbe8a5737e8647e/FirebaseInstallations/Source/Tests/Unit/FIRInstallationsSingleOperationPromiseCacheTests.m#L51)

Reply:
Added a test.